plugins/modules/eos_route_maps.py

```diff
@@ -0,0 +1,1404 @@
+  EOS platforms.
+version_added: 2.0.0
+author: Gomathi Selvi Srinivasan (@GomathiselviS)
+notes:
+- Tested against Arista EOS 4.23.0F
+- This module works with connection C(network_cli). See the L(EOS Platform Options,eos_platform_options).
+options:
+    config:
+      description: A list of route-map options
+      type: list
+      elements: dict
+      suboptions:
+        route_map_name:
+          description: Route map name.
+          type: str
+        entries:
+          description: Route Map entries.
+          type: list
+          elements: dict
+          suboptions:
+            statement:
+              description: statement name
+              type: str
+            source:
+              description: Rename/Copy configuration
+              type: dict
+              suboptions:
+                action:
+                  description: rename or copy configuration
+                  type: str
+                  choices: ["rename", "copy"]
+                source_map_name:
+                  description: Source route map name.
+                  type: str
+                overwrite:
+                  description: if True, overwrite existing config.
+                  type: bool
+            action:
+              description: Action for matching routes
+              type: str
+              choices: ["deny", "permit"]
+            sequence:
+              description: Index in the sequence.
+              type: int
+            continue_map:
+              description: Route map entry sequence number.
+              type: int
+            description:
+              description: Description for the route map.
+              type: str
+            sub_route_map:
+              description: Sub route map
+              type: dict
+              suboptions:
+                name:
+                  description: sub route map name
+                  type: str
+                invert_result:
+                  description: Invert sub route map result
+                  type: bool
+            set:
+              description: set route attributes.
+              type: dict
+              suboptions:
+                as_path:
+                  description: Set as-path.
+                  type: dict
+                  suboptions:
+                    match:
+                      description: Match the entire as-path.
+                      type: dict
+                      suboptions:
+                        as_number:
+                          description: as number to use (includes auto;in csv format)
+                          type: str
+                        none:
+                          description: Remove matching AS numbers
+                          type: bool
+                    prepend:
+                      description: Prepend to the as-path.
+                      type: dict
+                      suboptions:
+                        as_number:
```

Comment:
here, `as_number` should be list of str as it supports list

Reply:
I obtain the value in csv format, and handle it inside rm_templates.